### PR TITLE
fix(typescript-build): set logger instance correctly

### LINKED ISSUE #504: Consola got RangeError because of circular structure conversion in @nuxt/typescript-build
**Describe the bug**

Trying to create an Nuxt module for custom `consola` reporter using `winston` in my Nuxt app, then I got an error below when I run `yarn dev`:

```
error: RangeError: Maximum call stack size exceeded
    at mergeObject (/path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:79:21)
    at deepmerge (/path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:117:10)
    at cloneUnlessOtherwiseSpecified (/path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:34:5)
    at /path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:94:23
    at Array.forEach (<anonymous>)
    at mergeObject (/path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:86:18)
    at deepmerge (/path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:117:10)
    at cloneUnlessOtherwiseSpecified (/path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:34:5)
    at /path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:94:23
    at Array.forEach (<anonymous>)
    at mergeObject (/path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:86:18)
    at deepmerge (/path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:117:10)
    at cloneUnlessOtherwiseSpecified (/path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:34:5)
    at /path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:94:23
    at Array.forEach (<anonymous>)
    at mergeObject (/path/to/nuxt-app/node_modules/deepmerge/dist/cjs.js:86:18)","args":[]}
```

tl;dr It was because of `fork-ts-checker-webpack-plugin`. It uses `deepmerge` [to merge option objects](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/blob/v6.1.0/src/ForkTsCheckerWebpackPlugin.ts#L46) since [v5.0.0](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/blob/v5.0.0/src/ForkTsCheckerWebpackPlugin.ts#L33) and `deepmerge` [doesn't support circular structure](https://github.com/TehShrike/deepmerge/issues/207#issuecomment-711061110) which `@nuxt/typescript-build` [depends](https://github.com/nuxt/typescript/blob/%40nuxt/typescript-runtime%402.0.2/packages/typescript-build/src/index.ts#L84).

I don't know how this should be fixed, but for now, I cannot create any custom loggers using JSON serialization.

Hopefully my English is good enough to explain the situation. Thanks.

**To Reproduce**
1. Create a custom `consola` reporter Nuxt module depending on `JSON.stringify` (like `consola.JSONReporter`)
2. Run the Nuxt application
3. See error

```ts
import consola from 'consola'
consola.setReporters([
  new consola.JSONReporter(),
])
```

**Expected behavior**
`RangeError` should not be occured with custom Consola Reporters.

### PR DESCRIPTION
**Pull request by rchl**
Instead of passing whole Consola instance to `fork-ts-checker-webpack-plugin` options, which can't be serialized and would crash when trying to use JSON.serialize on it (see how it can happen in the linked issue), pass only an interface with required methods.

Our custom `consola` was not used AT ALL before this change. It was not set in the correct place so `fork-ts-checker-webpack-plugin` ignored it completely and just logged errors using `console` API. So this is also fixed now, in addition to fixing the crash.

Fixes #504

### CODE REVIEW

**Comment by rchl** [comment]
Actually, I will change it a bit to align better with current logic as it's not exactly as I've stated above.

`fork-ts-checker-webpack-plugin` creates `silent` logger by default for `infrastructure` and uses `console` for `issues`.

**Review by danielroe — APPROVED** [review]
